### PR TITLE
Add login checking logic

### PR DESCRIPTION
**Pull request by bananapizzza**

### CODE REVIEW

File: data/user_list.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "users": [
+    {
+      "username": "abcde",
+      "password": "11111"
```
Comment (xaveng) [inline data/user_list.json:5]: We normally store hash values of password, not the password itself. Let's just have a look at some hash functions like md5.

File: src/backendLogic.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+const express = require('express');
+const port = 5000;
```
Comment (xaveng) [inline src/backendLogic.js]: Let's pull this below import statements

File: src/backendLogic.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        //Select workout randomly from the array
+        const selectedWorkout = workoutList[Math.floor(Math.random() * workoutList.length)];
+        res.send(JSON.stringify(selectedWorkout));
+    });
+});
+
+app.post('/check_login_info', (req, res) => {
+    fs.readFile(`${path.parse(appRoot).dir}/data/user_list.json`, 'utf-8', (err, data) => {
+        const userList = JSON.parse(data).users;
+        let isValidUser = false;
```
Comment (xaveng) [inline src/backendLogic.js]: let's change it to try-catch form

File: src/backendLogic.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+const app = express();
+const path = require('path');
+const appRoot = path.resolve(__dirname);
+const cors = require('cors');
+
+app.use(express.json());
+app.use(cors());
+
+app.get('/', (req, res) => {
+    fs.readFile(`${path.parse(appRoot).dir}/data/workout_list.json`, 'utf-8', (err, data) => {
```
Comment (xaveng) [inline src/backendLogic.js]: let's change this with async module.
You can change it to something like below
```
const fileContent = await fs.readFileSync(`path`);
JSON.parse(fileContent);
....
```

Comment (xaveng) [reply src/backendLogic.js]: This should have look like this
```
fs.readFile(path, callback);
->
Promise(fs.readFile(path)).then(callback);
->
const fileContent = await fs.readFileSync(`path`);
```
Please note that you need to specify `async` to the function definition in order to use `await` keyword.

Comment (xaveng) [reply src/backendLogic.js]: fyi: https://blog.risingstack.com/mastering-async-await-in-nodejs/

Comment (xaveng) [reply src/backendLogic.js]: https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original

You can also change callback type of function into async-await usable function

**Review by xaveng — COMMENTED** [review]
good job

File: data/user_list.json (hunk 1 above)
Comment (bananapizzza) [reply data/user_list.json:5]: Okay

File: src/backendLogic.js (hunk 2 above)
Comment (bananapizzza) [reply src/backendLogic.js]: Changed

File: src/backendLogic.js (hunk 4 above)
Comment (bananapizzza) [reply src/backendLogic.js]: Changed

**Pull request merged**